google-cloud-spanner/src/main/java/com/google/cloud/spanner/DelayedMultiplexedSessionTransaction.java

```diff
@@ -119,4 +121,28 @@ public ReadOnlyTransaction readOnlyTransaction(TimestampBound bound) {
                     .readOnlyTransaction(bound),
             MoreExecutors.directExecutor()));
   }
+
+  @Override
+  public CommitResponse writeAtLeastOnceWithOptions(
```

Comment:
This method seems to be blocking. The methods in this class are intended to be non-blocking.

EDIT: When I come to think of it, we don't really need this one to be non-blocking, as the only way that this is invoked is through `writeAtLeastOnce(..)`. And that method again is blocking.

So would you mind:
1. Adding a comment to the method explaining why it is OK to implement this as a blocking method.
2. Maybe simplify the method a bit. We don't need the `ApiFutures.transform(...)` construction. Just call `this.sessionFuture.get()` at the start of the method, and then return `new MultiplexedSessionTransaction(..).writeAtLeastOnceWithOptions(...)` using the session reference that you get from the `this.sessionFuture.get()` call.

Reply:
Done.